### PR TITLE
build: fix tarantool build failure on xcode 12.5

### LINKED ISSUE #6076: Build failure on XCode 12.5
**Bug description**

After XCode 12.5 upgrade build started to fail in the following way:

```
In file included from /Users/s.ostanevich/workspaces/t.build/src/lib/core/backtrace.cc:42:
In file included from /Users/s.ostanevich/workspaces/t.build/src/lib/core/assoc.h:53:
In file included from /Users/s.ostanevich/workspaces/t.build/src/lib/salad/mhash.h:66:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.3.sdk/usr/include/c++/v1/math.h:309:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.3.sdk/usr/include/c++/v1/type_traits:418:
/Users/s.ostanevich/workspaces/t.build/src/lib/small/version:1:1: error: expected unqualified-id
1.0.0-19-gc1b6abd
^
```
The reason is `src/lib/small/VERSION` file is treated as C++ standard library <version> on a filesystem with case-insensitive names. 

To avoid the problems appearance in the future we need to update

a) include paths across the project to avoid the too-high include path with includes as `#include "small/region.h"`
b) the includes themselves to match the a) part

Similar problem appeared at the top of Tarantool repo, where another VERSION file is present. 

* OS: [e.g. Linux/MacOS/FreeBSD/OpenBSD]
MacOS

* OS Version: [e.g. Debian 11, FreeBSD 11]
BigSur 11.2.3, XCode 12.5

* Architecture: [amd64, arm]
AMD64

Place `tarantool --version` output here.
```
$ git describe
2.9.0-26-g75b2ccce8
```

**Steps to reproduce**
`cmake . && make`

### PR DESCRIPTION
**Pull request by EvgenyMekhanik**
`VERSION` files in small subproject and in tarantool are treated as C++
standard library on a filesystem with case-insensitive names. So we have
to delete the root of tarantool project from `include_directories` in
tarantool CMake. Also we have to change `include_directories` in tarantool
CMake from the root of `small` project to `small` subfolder in `small` project.
Closes #6076

### CODE REVIEW

**Comment by sergos** [comment]
Please add an explanation for `crc32.c` rename in commit message. 
Also, the name selection is questionable. The src/crc32.c mentions `architecture-specific implementation`, so perhaps better choice is `crc32_impl.c`

File: src/box/alter.cc [hunk 1]
```diff
@@ -44,7 +44,7 @@
 #include "tuple.h"
 #include "fiber.h" /* for gc_pool */
 #include "scoped_guard.h"
-#include "third_party/base64.h"
+#include "base64.h"
```
Comment (sergos) [inline src/box/alter.cc:47]: It looks like you wanted to update all `#include ""` for `#include <>` across the patch but some are slipped. Or is there more complex logic behind?

**Review by sergos — CHANGES_REQUESTED** [review]

File: src/box/alter.cc (hunk 1 above)
Comment (EvgenyMekhanik) [reply src/box/alter.cc]: Fixed some places some places need `<>`, in other places, I fixed it by analogy in the entire patch

File: CMakeLists.txt [hunk 2]
```diff
@@ -283,10 +283,10 @@ find_package_message(TARANTOOL_VERSION
 include_directories(${PROJECT_SOURCE_DIR}/src)
 include_directories(${PROJECT_BINARY_DIR}/src)
 include_directories(${PROJECT_SOURCE_DIR}/src/lib)
-include_directories(${PROJECT_SOURCE_DIR}/src/lib/small)
+include_directories(${PROJECT_SOURCE_DIR}/src/lib/small/small)
```
[the remark this answers is not in the document]
Comment (EvgenyMekhanik) [reply CMakeLists.txt]: Fixed

**Comment by EvgenyMekhanik** [comment]
> Please add an explanation for `crc32.c` rename in commit message.
> Also, the name selection is questionable. The src/crc32.c mentions `architecture-specific implementation`, so perhaps better choice is `crc32_impl.c`

No need to changed crc32.c only crc32.h. I change crc32.c by analogy. After you comment i changed crc32.c back. And i changed crc32.h to crc32_impl.h as you recommended.

File: src/lib/core/decimal.c [hunk 3]
```diff
@@ -30,8 +30,8 @@
  */
 
 #include "decimal.h"
-#include "third_party/decNumber/decContext.h"
-#include "third_party/decNumber/decPacked.h"
+#include <decNumber/decContext.h>
+#include <decNumber/decPacked.h>
```
Comment (sergos) [inline src/lib/core/decimal.c:34]: Shall we go deeper in this particular case and update include path to third_party/decNumber?

File: src/lib/core/fiber.c [hunk 4]
```diff
@@ -152,7 +152,7 @@ cpu_stat_end(struct cpu_stat *stat, struct clock_stat *cord_clock_stat)
 
 #endif /* ENABLE_FIBER_TOP */
 
-#include "third_party/valgrind/memcheck.h"
+#include <valgrind/memcheck.h>
```
Comment (sergos) [inline src/lib/core/fiber.c:155]: ditto

File: src/lua/httpc.c [hunk 5]
```diff
@@ -35,7 +35,7 @@
 #define DRIVER_LUA_UDATA_NAME	"httpc"
 
 #include "http_parser/http_parser.h"
-#include "src/httpc.h"
+#include "../httpc.h"
```
Comment (sergos) [inline src/lua/httpc.c:38]: wha? why that?

File: src/lua/init.c [hunk 6]
```diff
@@ -51,8 +51,8 @@
 #include "lua/errno.h"
 #include "lua/socket.h"
 #include "lua/utils.h"
-#include "third_party/lua-cjson/lua_cjson.h"
-#include "third_party/lua-yaml/lyaml.h"
+#include <lua-cjson/lua_cjson.h>
+#include <lua-yaml/lyaml.h>
```
Comment (sergos) [inline src/lua/init.c:55]: same question - can we have a list of include folders rather include names with folders?

File: src/crc32.c [hunk 7]
```diff
@@ -31,7 +31,7 @@
 #include "crc32.h"
 
 #include <trivia/config.h>
-#include <third_party/crc32.h>
+#include <crc32_impl.h>
```
Comment (sergos) [inline src/crc32.c:34]: I believe we'll have crc32_impl.h and crc32.c in third_party folder after rename only first one?
Is it consistent? For me it looks natural if header has the same name as the code file.

File: src/lua/httpc.c (hunk 5 above)
Comment (EvgenyMekhanik) [reply src/lua/httpc.c]: Fixed

File: src/crc32.c (hunk 7 above)
Comment (EvgenyMekhanik) [reply src/crc32.c:34]: Fixed

**Review by sergos — APPROVED** [review]

**Review by iskander232 — APPROVED** [review]
LGTM

**Comment by Korablev77** [comment]
I tried to build on fresh Mac and I have to apply this diff to make build succeed:
```
diff --git a/third_party/lua-cjson/strbuf.c b/third_party/lua-cjson/strbuf.c
index 69057ffb3..87cdda64c 100644
--- a/third_party/lua-cjson/strbuf.c
+++ b/third_party/lua-cjson/strbuf.c
@@ -28,7 +28,7 @@
 #include <string.h>
 
 #include "strbuf.h"
-#include "small/ibuf.h"
+#include "ibuf.h"
 
 static void die(const char *fmt, ...)
 {
```

**Comment by EvgenyMekhanik** [comment]
> I tried to build on fresh Mac and I have to apply this diff to make build succeed:
> 
> ```
> diff --git a/third_party/lua-cjson/strbuf.c b/third_party/lua-cjson/strbuf.c
> index 69057ffb3..87cdda64c 100644
> --- a/third_party/lua-cjson/strbuf.c
> +++ b/third_party/lua-cjson/strbuf.c
> @@ -28,7 +28,7 @@
>  #include <string.h>
>  
>  #include "strbuf.h"
> -#include "small/ibuf.h"
> +#include "ibuf.h"
>  
>  static void die(const char *fmt, ...)
>  {
> ```

This is a new header in this file, on the moment of may patch it was no such header)

**Comment by Totktonada** [comment]
It would be good to reflect the new changes in the PR description (say, just copy the new commit message here).

It would be good to link the small's PR from this one.

No other objections from me.

**Review by Totktonada — APPROVED** [review]

**Pull request merged**